### ISSUE TITLE
Error on Go get and acs-engine build

### ISSUE THREAD

**Issue by mfaizanse** [Azure/acs-engine#1068]
**Is this a request for help?**:
YES
---

**Is this an ISSUE or FEATURE REQUEST?** (choose one):
ISSUE
---

**What version of acs-engine?**:
commit 42b4fa3  
---

**Orchestrator and version (e.g. Kubernetes, DC/OS, Swarm)**
Kubernetes

**What happened**:
acs-engine failed to get and build

**What you expected to happen**:
Build success

**How to reproduce it** (as minimally and precisely as possible):
https://github.com/Azure/acs-engine/blob/master/docs/acsengine.md#downloading-and-building-acs-engine

**Anything else we need to know**:
![image](https://user-images.githubusercontent.com/9897945/28460822-4006b958-6e2d-11e7-9682-135b03ed5f84.png)

If I deploy a kubernetes cluster using azure portal then it deploy using the kubernetes 1.6.6 version and that results in dns resolutoin failure and No connectivity inside containers. And when I was tried to deploy using the acs-engine then this issue came up.

**Comment by weinong** [comment]
have you tried
`make prereqs`
`make build`

**Comment by mfaizanse** [comment]
@weinong I am on windows?

**Comment by mfaizanse** [comment]
@weinong Where should I run these commands? I have tried in "C:\gopath\src\github.com\Azure\acs-engine" but its failing.

![image](https://user-images.githubusercontent.com/9897945/28512333-eaf67ff8-706a-11e7-8d6f-d8f877df16d2.png)

**Comment by amanohar** [comment]
@mfaizanse you need to install make (since you are running this on windows) or use the instructions at: https://github.com/Azure/acs-engine/blob/master/docs/acsengine.md to mount the repo in a container.

**Comment by mfaizanse** [comment]
@amanohar I installed make and tried but failed. Please see my above comment.

**Comment by amanohar** [comment]
@mfaizanse from the link I shared above can you try mounting your repo into a container by running: Windows (PowerShell): .\scripts\devenv.ps1 and then trying make prepreqs? Easiest way to start developing on ACS Engine is to use Docker.

**Comment by mfaizanse** [comment]
@amanohar I have tried the container method and its working fine.

**Comment by seanknox** [comment]
I can confirm this is an issue when cloning the repo via `go get`:

```
$ go get github.com/seanknox/acs-engine
# github.com/Azure/acs-engine/pkg/acsengine
acs-engine/pkg/acsengine/engine.go:185: undefined: Asset
acs-engine/pkg/acsengine/engine.go:234: undefined: Asset
acs-engine/pkg/acsengine/engine.go:288: undefined: Asset
acs-engine/pkg/acsengine/engine.go:1178: undefined: Asset
acs-engine/pkg/acsengine/engine.go:1211: undefined: Asset
acs-engine/pkg/acsengine/engine.go:1233: undefined: Asset
acs-engine/pkg/acsengine/engine.go:1259: undefined: Asset
acs-engine/pkg/acsengine/engine.go:1299: undefined: Asset
```

**Comment by Teodelas** [comment]
I'm having the same issue on windows.

**Comment by orkun1675** [comment]
Also having this issue on Windows and WSL.

**Comment by seanknox** [comment]
Figured out what's causing this. We used to check in `pkg/acsengine/templates.go` and recently removed it from the repo as it's a build artifact and was causing a lot of developer friction.

`go get` attempts a build but first `pkg/acsengine/templates.go` needs to be generated via `make generate`. Since it doesn't exist when cloning, cloning will always error out as it attempts to build without the generated file in place. While ugly and something we should fix, this doesn't have any functional effect. Looking into how we can fix this without checking in `pkg/acsengine/templates.go` again.
